pkg/util/util.go

```diff
@@ -1164,17 +1167,35 @@ func GetAWSRegion() (string, error) {
 		"af-south-1":     "af-south-1",
 		"il-central-1":   "il-central-1",
 	}
-	nodesList := &corev1.NodeList{}
-	if ok := KubeList(nodesList); !ok || len(nodesList.Items) == 0 {
-		return "", fmt.Errorf("Failed to list kubernetes nodes")
+	var awsRegion string
+	infrastructure := &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+	}
+	if _, _, err := KubeGet(infrastructure); err != nil {
+		log.Infof("Failed to fetch cluster infrastructure details: %v", err)
+	}
+	if infrastructure.Status.PlatformStatus != nil && infrastructure.Status.PlatformStatus.AWS != nil {
```

Comment:
log a warn message if we don't find the expected details in the infrastructure object, before falling back to node name.

Reply:
Done, Thanks.